packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -378,6 +394,61 @@ class RenderEditable extends RenderBox {
     return newOffset;
   }
 
+  // Handles shortcut functionality including cut, copy, paste and select all
+  // using control + (X, C, V, A).
+  void _handleShortcuts(bool ctrl, bool xKey, bool vKey, bool cKey, bool aKey) async {
+    if (cKey && !selection.isCollapsed) {
+      Clipboard.setData(new ClipboardData(text: selection.textInside(text.text)));
+    } else if (xKey && !selection.isCollapsed) {
+      Clipboard.setData(new ClipboardData(text: selection.textInside(text.text)));
+      _textSelectionDelegate.textEditingValue = new TextEditingValue(
+        text: selection.textBefore(text.text)
+            + selection.textAfter(text.text),
+        selection: new TextSelection.collapsed(offset: selection.start),
+      );
+    } else if (vKey) {
+      // Snapshot the input before using `await`.
+      // See https://github.com/flutter/flutter/issues/11427
+      final TextEditingValue value = _textSelectionDelegate.textEditingValue;
+      final ClipboardData data = await Clipboard.getData(Clipboard.kTextPlain);
+      if (data != null) {
+        _textSelectionDelegate.textEditingValue = new TextEditingValue(
+          text: value.selection.textBefore(value.text)
+              + data.text
+              + value.selection.textAfter(value.text),
+          selection: new TextSelection.collapsed(
+              offset: value.selection.start + data.text.length
+          ),
+        );
+      }
+    } else if (aKey) {
+      _baseOffset = 0;
+      _extentOffset = _textSelectionDelegate.textEditingValue.text.length;
+      onSelectionChanged(
+        new TextSelection(
+            baseOffset: 0,
+            extentOffset: _textSelectionDelegate.textEditingValue.text.length,
```

Comment:
trivial nit: you're inconsistent about the indent level - you use 4 here, but 2 below. We should use 2 everywhere for indented arguments.